### ISSUE TITLE
lower case import path for github.com/sirupsen/logrus

### ISSUE THREAD

**Issue by pohly** [moby/moby#37281]
**Description**

All Docker .go files import github.com/Sirupsen/logrus, which according to https://github.com/sirupsen/logrus/blob/master/README.md is not the right name. After some confusion, all lower case (github.com/Sirupsen/logrus) has been chosen and other projects have adopted that, including several projects that Docker depends on (see for instance https://github.com/Microsoft/hcsshim/commit/461d6cb5453df54ec1b746a43dd2bb320f0f18f8).

Not doing the same makes it impossible to use both Docker libraries and recent versions of those other projects at the same time and locks Docker itself into using out-dated releases.

This was already mentioned in https://github.com/moby/moby/issues/33813#issuecomment-313819571 but that issue is about using logrus at all.

**Steps to reproduce the issue:**
1. set up a Go project that imports github.com/docker/docker and k8s.io/kubernetes/test/e2e/framework
2. run "dep init" + "dep ensure"

**Describe the results you received:**

dep detects the case mismatch and fails to resolve dependencies.

**Comment by vdemeester** [comment]
@pohly if you look at master we're using `sirupsen` (we did migrate about a year ago). The problems lies in the fact that we don't publish tags in this repos starting from [`v17.03.2-ce`](https://github.com/moby/moby/releases) —  by default `dep` will take the most recent tags (in the SemVer sense), and thus is getting a version of the sources that do not have this *migration* in.

A possible workaround to this is to use `[[override]]` in the `Gopkg.toml` file : 

```
[[override]]
  name = "github.com/docker/docker"
revision = "7848b8beb9d38a98a78b75f78e05f8d2255f9dfe"
```

That said it's definitely not optimal, and I really think (and advocate for) that we should tag versions on `moby/moby`, even if it's independent of the `docker-ce` distribuiton.

cc @tiborvass @thaJeztah @andrewhsu

**Comment by thaJeztah** [comment]
Never mind, I just see you mentioned that; highest docker release that was done from this repo was 17.05 (but likely tagged before the 17.03.2 release)

I'm not sure what's best for tagging; existing tags are mainly an archive of past docker releases (before the introduction of Moby as a project), so we can't remove those, and I don't know if just adding automated (weekly?) tags would give much value, other than address the dep issue.

Note that current Docker's releases are done from a repository in which the "engine" and "cli" components are combined (https://github.com/docker/docker-ce/), which makes it not usable for vendoring, but there will be some changes in the release process, where an intermediate repository is used for the "engine" component, in which tags and release-branches can be created (the repository is created, but not yet in active use; https://github.com/docker/engine)

**Comment by pohly** [comment]
Indeed, the issue is fixed in master, which "dep" doesn't pick up. Sorry, I missed that.

The expectation in dep is that projects follow semantic versioning (https://semver.org/) and tag their releases. Because this isn't happening in the moby/moby repo, the only option one has is to manually pick a specific revision or go with the master branch. Both is bad: the former locks the project to a certain revision and will probably miss important bug fixes, the latter increases the chance of random API and thus build breakages.

**Comment by pohly** [comment]
... and indeed, the latest revision from the master branch doesn't work with Kubernetes 1.10.2, because github.com/docker/docker/pkg/jsonlog is missing.

**Comment by pohly** [comment]
Sebastiaan van Stijn <notifications@github.com> writes:
> Never mind, I just see you mentioned that; highest docker release that
> was done from this repo was 17.05 (but likely tagged before the
> 17.03.2 release)

Beware that dep (or rather, https://github.com/Masterminds/semver) does
not consider v17.05.0-ce as a regular release tag because of the -ce
suffix.
